lib/api/src/js/export.rs

```diff
@@ -57,6 +110,29 @@ impl VMGlobal {
     pub(crate) fn new(global: Global, ty: GlobalType) -> Self {
         Self { global, ty }
     }
+
+    /// Saves the global value into the snapshot
+    pub fn save_snapshot(&self, index: usize, snapshot: &mut StoreSnapshot) {
+        if let Some(val) = self.global.as_f64() {
+            let entry = snapshot.globals.entry(index as u32).or_default();
+            *entry = val as u128;
+        }
+    }
+
+    /// Restores the global value from the snapshot
+    pub fn restore_snapshot(&mut self, index: usize, snapshot: &StoreSnapshot) {
+        let index = index as u32;
+        if let Some(entry) = snapshot.globals.get(&index) {
+            if let Some(existing) = self.global.as_f64() {
+                let existing = existing as u128;
```

Comment:
this looks a bit mysterious to me. Are you sure you compare comparable things here (f64->u128 vs mem->u128 or did I miss something)?